pkg/spanconfig/spanconfigmanager/manager.go

```diff
@@ -0,0 +1,147 @@
+// Copyright 2021 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package spanconfigmanager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/cockroach/pkg/jobs"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/kv"
+	"github.com/cockroachdb/cockroach/pkg/security"
+	"github.com/cockroachdb/cockroach/pkg/spanconfig"
+	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
+	"github.com/cockroachdb/cockroach/pkg/sql/sessiondata"
+	"github.com/cockroachdb/cockroach/pkg/sql/sqlutil"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/cockroach/pkg/util/stop"
+)
+
+// Manager is responsible for ensuring that one (and only one) span config
+// reconciliation job is ever running. It also provides the job with all the
+// components it needs to actually perform reconciliation.
+type Manager struct {
+	db      *kv.DB
+	jr      *jobs.Registry
+	ie      sqlutil.InternalExecutor
+	knobs   *spanconfig.TestingKnobs
+	stopper *stop.Stopper
+}
+
+var _ spanconfig.ReconciliationDependencies = &Manager{}
+
+// New constructs a new Manager.
+func New(
+	db *kv.DB,
+	jr *jobs.Registry,
+	ie sqlutil.InternalExecutor,
+	stopper *stop.Stopper,
+	knobs *spanconfig.TestingKnobs,
+) *Manager {
+	if knobs == nil {
+		knobs = &spanconfig.TestingKnobs{}
+	}
+	return &Manager{
+		db:      db,
+		jr:      jr,
+		ie:      ie,
+		knobs:   knobs,
+		stopper: stopper,
+	}
+}
+
+// StartJobIfNoneExists will create and start the span config reconciliation job
+// iff it hadn't been created already. Returns a boolean indicating if the job
+// was started.
+func (m *Manager) StartJobIfNoneExists(ctx context.Context) bool {
```

Comment:
I'm not sure that this synchronous method is really that helpful. What if we just had this thing sit in a loop and periodically, like every hour or something (after startup) go and run the check. If you need a synchronous hook, you can have it poke the loop or something. My guess is that the synchronous thing is handy in testing. Something smells about it being called in preStart.